extensions/interactions/TextInput/TextInput.js

```diff
@@ -81,3 +81,21 @@ oppia.directive('oppiaShortResponseTextInput', [
     };
   }
 ]);
+
+oppia.factory('textInputRulesService', [function() {
+  return {
+    Equals: function(answer, inputs) {
+      return answer.toLowerCase() == inputs.x.toLowerCase();
+    },
+    CaseSensitiveEquals: function(answer, inputs) {
+      return answer == inputs.x;
+    },
+    StartsWith: function(answer, inputs) {
+      return answer.indexOf(inputs.x, 0) == 0;
+    },
+    Contains: function(answer, inputs) {
+      return answer.toLowerCase().indexOf(inputs.x.toLowerCase(), 0) != -1;
+    }
+
```

Comment:
Just a clarification: It seems like the python FuzzyMatches implementation is case-sensitive such that case differences contribute to distance, but yet there is a check at the start to see if they are equivalent in lower case. Is this the intended behavior?